client/src/partials/reports/balance_mensuelle/balance_mensuelle.js

```diff
@@ -58,6 +108,8 @@ angular.module('bhima.controllers')
       session.sumOldCredit = $scope.balance_mensuelle.data.reduce(function sum(a, b) { return a + b.old_credit; },0);
       session.sumDebit = $scope.balance_mensuelle.data.reduce(function sum(a, b) { return a + b.debit; },0);
       session.sumCredit = $scope.balance_mensuelle.data.reduce(function sum(a, b) { return a + b.credit; },0);
+      session.solde_debit = $scope.balance_mensuelle.data.reduce(function sum(a, b) { return a + b.solde_debit; },0);
+      session.solde_credit = $scope.balance_mensuelle.data.reduce(function sum(a, b) { return a + b.solde_credit; },0);      
```

Comment:
There are six (!) loops through `$scope.balance_mensuelle.data` going on here. Here is a trick to reduce it to a single loop:
```js
var sums = {
  sumOldDebit : 0,
  sumOldCredit : 0,
  sumDebit : 0,
  sumCredit : 0,
  solde_debit : 0,
  solde_credit :0
};

sums = $scope.balance_mensuelle.data.reduce(function (summer, row) {
  summer.sumOldDebit += row.old_debit;
  summer.sumOldCredit += row.old_credit;
  summer.sumDebit += row.debit;
  summer.sumCredit += row.credit;
  summer.solde_debit += row.solde_debit;
  summer.solde_credit += row.solde_credit;
  return summer;
}, sums);

// Everything is now summed in one loop in `sums`!  You can attach `sums` to `session` as you want.
```